test/integration/visual_shininess.cc

```diff
@@ -0,0 +1,120 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <mutex>
+#include <functional>
+
+#include <ignition/math/Rand.hh>
+
+#include "test_config.h"
+
+#include "gazebo/physics/physics.hh"
+#include "gazebo/sensors/sensors.hh"
+#include "gazebo/common/Timer.hh"
+#include "gazebo/rendering/Camera.hh"
+#include "gazebo/sensors/CameraSensor.hh"
+
+#include "gazebo/test/ServerFixture.hh"
+
+using namespace gazebo;
+class VisualShininess : public RenderingFixture
+{
+};
+
+/////////////////////////////////////////////////
+void CheckShininessService(const std::string &_modelName,
+    double _expectedShininess)
+{
+  std::string serviceName = '/' + _modelName + "/shininess";
+  ignition::transport::Node ignNode;
+
+  {
+    std::vector<ignition::transport::ServicePublisher> publishers;
+    EXPECT_TRUE(ignNode.ServiceInfo(serviceName, publishers));
+  }
+
+  ignition::msgs::StringMsg request;
+  gazebo::msgs::Any reply;
+  const unsigned int timeout = 3000;
+  bool result = false;
+  request.set_data(_modelName);
+  EXPECT_TRUE(ignNode.Request(serviceName, request, timeout, reply, result));
+  EXPECT_TRUE(result);
+  EXPECT_EQ(msgs::Any_ValueType_DOUBLE, reply.type());
+  EXPECT_DOUBLE_EQ(_expectedShininess, reply.double_value()) << _modelName;
+}
+
+/////////////////////////////////////////////////
+TEST_F(VisualShininess, ShapesShininessServices)
+{
+  Load("worlds/shapes_shininess.world", true);
+
+  CheckShininessService("ground_plane", 0.0);
+  CheckShininessService("box", 1.0);
+  CheckShininessService("sphere", 5.0);
+  CheckShininessService("cylinder", 10.0);
+}
+
+/////////////////////////////////////////////////
+TEST_F(VisualShininess, ShapesShininess)
+{
+  Load("worlds/shapes_shininess.world");
+
+  // Make sure the render engine is available.
+  if (rendering::RenderEngine::Instance()->GetRenderPathType() ==
+      rendering::RenderEngine::NONE)
+  {
+    gzerr << "No rendering engine, unable to run camera test"
+          << std::endl;
+    return;
+  }
+  gazebo::rendering::ScenePtr scene = gazebo::rendering::get_scene();
+  ASSERT_NE(nullptr, scene);
+
+  // Wait until all models are inserted
+  unsigned int sleep = 0;
+  unsigned int maxSleep = 30;
+  rendering::VisualPtr box, sphere, cylinder;
+  while ((!box || !sphere || !cylinder) && sleep < maxSleep)
+  {
+    event::Events::preRender();
+    event::Events::render();
+    event::Events::postRender();
+
+    box = scene->GetVisual("box");
+    cylinder = scene->GetVisual("cylinder");
+    sphere = scene->GetVisual("sphere");
+
+    common::Time::MSleep(100);
+    sleep++;
+  }
+  EXPECT_TRUE(scene->Initialized());
```

Comment:
Should we add assertions that `box`, `cylinder`, and `sphere` are not `nullptr`? In the case that `sleep >= maxSleep` and it exits the loop